CDCgov/prime-reportstream
# 1124 HL7 ingest date type support

This PR adds support for the DATE schema type along with the DT (date) HL7 type. 

To test:
Run the ObservationMessageTests or just `./gradlew test` and verify the tests are successfully.

## Changes
- Support schema DATE type conversion from an HL7 timestamp and provide proper warnings of precision in the data.
- Support HL7 DT date decoding
- Added unit tests and updated data tests

## Checklist

### Testing

### Security

### Process

## Security
*List potential security threats and mitigations in the PR*

## Fixes
- #1124

## To Be Done
